### PR TITLE
Phase5: Update the AccuracyScorer.score signature

### PR DESCRIPTION
**Pull request by sk-ip**
Update all the AccuracyScorer signature to take modelcontext and sourcescontext.

Addresses #732

Signed-off-by: Sudhanshu <Sudhanshukumar5459@gmail.com>

### CODE REVIEW

**Comment by pdxjohnny** [comment]
Good stuff! Just wanted to make a note here that we need to remember to

- [ ] Create a new `AccuracyScorer` based on the old NER `accuracy()` method

Correct me if I'm wrong because I didn't check around to see if you'd already done that, or if for some reason that doesn't need to be done. I just read the commits individually when reviewing this just now I didn't look at the codebase as a whole.

**Comment by sk-ip** [comment]
@pdxjohnny , yes for the ner model we need to do #866 , In the ner accuracy the spacy.scorer was used.

File: model/scikit/tests/test_scikit.py [hunk 1]
```diff
@@ -128,15 +129,13 @@ async def test_00_train(self):
...
-            async with sources() as sctx, model() as mctx, scorer() as actx:
-                res = await mctx.accuracy(sctx, actx)
-                if self.MODEL_TYPE is "CLUSTERING":
-                    self.assertTrue(res is not None)
-                elif self.MODEL_TYPE == "REGRESSION":
-                    self.assertTrue(0 <= res <= float("inf"))
-                else:
-                    self.assertTrue(0 <= res <= 1)
+        res = await accuracy(self.model, self.scorer, self.sources)
+        if self.MODEL_TYPE is "CLUSTERING":
```
Comment (pdxjohnny) [inline model/scikit/tests/test_scikit.py:133]:
```suggestion
        if self.MODEL_TYPE == "CLUSTERING":
```

File: dffml/accuracy/accuracy.py [hunk 2]
```diff
@@ -24,9 +22,7 @@ def __init__(self, parent: "Accuracy") -> None:
         self.parent = parent
 
     @abc.abstractmethod
-    async def score(
-        self, mctx: ModelContext, sources: SourcesContext,
-    ) -> float:
+    async def score(self, mctx, sources) -> float:
```
Comment (pdxjohnny) [inline dffml/accuracy/accuracy.py:25]: I think we can remove this commit

```console
$ git rebase -i HEAD~8
```

Then drop the commit.

Should be able to replace it with removal of `from ..accuracy.accuracy import AccuracyContext` line from `dffml/model/model.py`.

**Comment by pdxjohnny** [comment]
Very nice! I think we have some issues with the tests in `.`, it would be good to investigate those before doing the rebase, perhaps in an other PR.

```console
$ python -m unittest discover -v tests
```

**Comment by pdxjohnny** [comment]
```
======================================================================
ERROR: test_docstring (test_docstrings.high_level_accuracy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/dffml/dffml/tests/test_docstrings.py", line 209, in testcase
    run_doctest(obj, state)
  File "/home/runner/work/dffml/dffml/tests/test_docstrings.py", line 178, in run_doctest
    raise Exception(output.getvalue())
Exception: Trying:
    import asyncio
Expecting nothing
ok
Trying:
    from dffml import *
Expecting nothing
ok
Trying:
    model = SLRModel(
        features=Features(
            Feature("Years", int, 1),
        ),
        predict=Feature("Salary", int, 1),
        directory="tempdir",
    )
Expecting nothing
ok
Trying:
    async def main():
        print(
            "Accuracy:",
            await accuracy(
                model,
                {"Years": 4, "Salary": 50},
                {"Years": 5, "Salary": 60},
            ),
        )
Expecting nothing
ok
Trying:
    asyncio.run(main())
Expecting:
    Accuracy: 1.0
**********************************************************************
File "/home/runner/work/dffml/dffml/dffml/high_level.py", line 436, in accuracy
Failed example:
    asyncio.run(main())
Exception raised:
    Traceback (most recent call last):
      File "/usr/share/miniconda/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest accuracy[4]>", line 1, in <module>
        asyncio.run(main())
      File "/usr/share/miniconda/lib/python3.8/asyncio/runners.py", line 43, in run
        return loop.run_until_complete(main)
      File "/usr/share/miniconda/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
        return future.result()
      File "<doctest accuracy[3]>", line 4, in main
        await accuracy(
      File "/home/runner/work/dffml/dffml/dffml/high_level.py", line 440, in accuracy
        async with sources as sources, model as model, accuracy_scorer as accuracy_scorer:
    AttributeError: __aexit__


======================================================================
ERROR: test_docstring (test_docstrings.noasync_accuracy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/dffml/dffml/tests/test_docstrings.py", line 209, in testcase
    run_doctest(obj, state)
  File "/home/runner/work/dffml/dffml/tests/test_docstrings.py", line 178, in run_doctest
    raise Exception(output.getvalue())
Exception: Trying:
    from dffml import *
Expecting nothing
ok
Trying:
    from dffml.noasync import *
Expecting nothing
ok
Trying:
    model = SLRModel(
        features=Features(
            Feature("Years", int, 1),
        ),
        predict=Feature("Salary", int, 1),
        directory="tempdir",
    )
Expecting nothing
ok
Trying:
    def main():
        print(
            "Accuracy:",
            accuracy(
                model,
                {"Years": 4, "Salary": 50},
                {"Years": 5, "Salary": 60},
            ),
        )
Expecting nothing
ok
Trying:
    main()
Expecting:
    Accuracy: 1.0
**********************************************************************
File "/home/runner/work/dffml/dffml/dffml/noasync.py", line 79, in accuracy
Failed example:
    main()
Exception raised:
    Traceback (most recent call last):
      File "/usr/share/miniconda/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest accuracy[4]>", line 1, in <module>
        main()
      File "<doctest accuracy[3]>", line 4, in main
        accuracy(
      File "/home/runner/work/dffml/dffml/dffml/noasync.py", line 23, in accuracy
        return asyncio.run(high_level.accuracy(*args, **kwargs))
      File "/usr/share/miniconda/lib/python3.8/asyncio/runners.py", line 43, in run
        return loop.run_until_complete(main)
      File "/usr/share/miniconda/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
        return future.result()
      File "/home/runner/work/dffml/dffml/dffml/high_level.py", line 440, in accuracy
        async with sources as sources, model as model, accuracy_scorer as accuracy_scorer:
    AttributeError: __aexit__
```

**Comment by pdxjohnny** [comment]
We are seeing an issue with transformers, not sure if it's a versioning issue or not, re-running the CI in master to double check.

**Pull request merged**